packages/table/src/table-store.js

```diff
@@ -352,6 +352,8 @@ TableStore.prototype.updateColumns = function() {
 
   states.columns = [].concat(fixedLeafColumns).concat(leafColumns).concat(rightFixedLeafColumns);
   states.isComplex = states.fixedColumns.length > 0 || states.rightFixedColumns.length > 0;
+
+  this.states = {...states};
```

Comment:
We can use the [`merge` function](https://github.com/ElemeFE/element/blob/dev/src/utils/merge.js).

```js
this.states = merge({}, states)
```